app/src/main/java/com/edulectronics/tinycircuit/Models/Circuit.java

```diff
@@ -43,7 +45,13 @@ public Component getComponent(int i) {
     }
 
     public Component[] getAllComponents() {
-        return this.components;
+        ArrayList<Component> componentList = new ArrayList<>(Arrays.asList(components));
+        componentList.removeAll(Collections.singleton(null));
```

Comment:
Een array heeft geen removeAll methode (want java😞).
De removeAll(Collections.singleton(null)) zegt haal alles uit de verzameling wat gelijk is aan een lege collectie